exonum/src/messages2/mod.rs

```diff
@@ -0,0 +1,274 @@
+// Copyright 2018 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//!
+//! Message represents object received from p2p network.
+//! There next flow between objects:
+//! ```text
+//! +---------+           +---------------+                 +------------+
+//! | Vec<u8> |    ->     | SignedMessage |        ->       |  Protocol  |
+//! +---------+  (verify) +---------------+  (deserialize)  +------------+
+//!                 |                             |
+//!                 V                             V
+//!              (      message dropped if failed       )
+//!
+//! ```
+use hex::{FromHex, ToHex};
+use byteorder::{ByteOrder, LittleEndian};
+use failure::Error;
+
+use std::{borrow::Cow, cmp::PartialEq, fmt, mem, ops::Deref};
+
+use crypto::{hash, CryptoHash, Hash, PublicKey};
+use encoding;
+use storage::StorageValue;
+
+pub(crate) use self::{authorization::SignedMessage, helpers::HexStringRepresentation};
+pub use self::{helpers::BinaryForm, protocol::*};
+
+#[macro_use]
+mod compatibility;
+mod authorization;
+mod helpers;
+mod protocol;
+#[cfg(test)]
+mod tests;
+
+/// Version of the protocol. Different versions are incompatible.
+pub const PROTOCOL_MAJOR_VERSION: u8 = 1;
+
+/// Transaction raw buffer.
+/// This struct is used to transfer transactions in network.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct RawTransaction {
+    service_id: u16,
+    transaction_set: TransactionSetPart,
+}
+
+/// Concrete raw transaction transaction inside `TransactionSet`.
```

Comment:
There is no `TransactionSet`

Reply:
I agree, `TransactionSetPart` sounds a little bit weird.

Reply:
@vitvakatu `TransactionSet` is a set inside `blockchain` mod https://github.com/exonum/exonum/blob/master/exonum/src/blockchain/transaction.rs#L361

What did you mean that there is no `TransactionSet`?

@alekseysidorov  what your suggestions about this type?
This is a DTO of `TransactionSet` instance.